src/main/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcher.java

```diff
@@ -35,16 +36,19 @@ public CompositeSearchBasedFetcher(Set<SearchBasedFetcher> searchBasedFetchers,
 
     @Override
     public List<BibEntry> performSearch(String query) {
-        return fetchers.stream().flatMap(searchBasedFetcher -> {
+        List<BibEntry> result = fetchers.parallelStream().flatMap(searchBasedFetcher -> {
             try {
                 return searchBasedFetcher.performSearch(query).stream();
             } catch (FetcherException e) {
                 LOGGER.warn(String.format("%s API request failed", searchBasedFetcher.getName()), e);
                 return Stream.empty();
             }
-        }).parallel()
-          .limit(maximumNumberOfReturnedResults)
-          .collect(Collectors.toList());
+        }).limit(maximumNumberOfReturnedResults)
+                                        .collect(Collectors.toList());
```

Comment:
Can't the cleanup be done right after `.limit`? If it's not easy, maybe add `.forEach` after the `.collect` and initialize the converter before the result variable?

Reply:
Shouldn't this cleanup actually be removed / replaced by the import cleanup process?